process/transaction/baseProcess.go

```diff
@@ -124,6 +126,11 @@ func (txProc *baseTxProcessor) checkTxValues(tx *transaction.Transaction, acntSn
 		return process.ErrLowerNonceInTransaction
 	}
 
+	err := txProc.economicsFee.CheckValidityTxValues(tx)
+	if err != nil {
+		return err
+	}
+
```

Comment:
i would add the check here if there is enough money for the tx fee. then leave the code as it is.

	txFee := txProc.economicsFee.ComputeFee(tx)
	if stAcc.Balance.Cmp(txFee) < 0 {
		return fmt.Errorf("%w, has: %s, wanted: %s",
			process.ErrInsufficientFee,
			stAcc.Balance.String(),
			txFee.String(),
		)
	}

Reply:
My variant is more optimum as it checks for existing fee only if the total cost of the transaction is greater than the account's balance. Statistically, close to 100% processing transactions will only need to be checked on the total cost of the transaction as they will succeed.

Reply:
changed